src/test/java/ch/uzh/ifi/hase/soprafs22/screwyourneighborserver/entity/HandScoreTest.java

```diff
@@ -221,4 +213,132 @@ void evaluates_score_with_multiple_highest_cards() {
+    assertThat(handPlayer2.getNumberOfWonTricks(), is(0));
+    assertThat(handPlayer2.getPoints(), is(-1));
+    assertThat(handPlayer3.getNumberOfWonTricks(), is(1));
+    assertThat(handPlayer3.getPoints(), is(-1));
+  }
+
+  @Test
+  void evaluates_score_with_two_stacked_rounds() {
+    matchBuilder =
+        GameBuilder.builder("test", gameRepository, participationRepository, playerRepository)
+            .withParticipation(PLAYER_1)
+            .withParticipation(PLAYER_2)
+            .withParticipation(PLAYER_3)
+            .withMatch();
+
+    matchBuilderWithHands =
+        matchBuilder
+            .withHandForPlayer(PLAYER_1)
+            .withCards(ACE_OF_CLUBS, QUEEN_OF_CLUBS, JACK_OF_CLUBS, KING_OF_HEARTS)
+            .withAnnouncedScore(2)
+            .finishHand()
+            .withHandForPlayer(PLAYER_2)
+            .withCards(KING_OF_CLUBS, JACK_OF_SPADES, EIGHT_OF_CLUBS, QUEEN_OF_HEARTS)
+            .withAnnouncedScore(0)
+            .finishHand()
+            .withHandForPlayer(PLAYER_3)
+            .withCards(SEVEN_OF_CLUBS, ACE_OF_SPADES, QUEEN_OF_SPADES, EIGHT_OF_SPADES)
+            .withAnnouncedScore(2)
+            .finishHand();
+    Game game =
+        matchBuilderWithHands
+            .withRound()
+            .withPlayedCard(PLAYER_1, QUEEN_OF_CLUBS)
+            .withPlayedCard(PLAYER_2, JACK_OF_SPADES)
+            .withPlayedCard(PLAYER_3, SEVEN_OF_CLUBS)
+            .finishRound()
+            .withRound()
+            .withPlayedCard(PLAYER_1, ACE_OF_CLUBS)
+            .withPlayedCard(PLAYER_2, QUEEN_OF_HEARTS)
+            .withPlayedCard(PLAYER_3, ACE_OF_SPADES)
+            .finishRound()
+            .withRound()
+            .withPlayedCard(PLAYER_1, KING_OF_HEARTS)
+            .withPlayedCard(PLAYER_2, KING_OF_CLUBS)
+            .withPlayedCard(PLAYER_3, EIGHT_OF_SPADES)
+            .finishRound()
+            .withRound()
+            .withPlayedCard(PLAYER_1, JACK_OF_CLUBS)
+            .withPlayedCard(PLAYER_2, EIGHT_OF_CLUBS)
+            .withPlayedCard(PLAYER_3, QUEEN_OF_SPADES)
+            .finishRound()
+            .finishMatch()
+            .build();
+
+    gameRepository.saveAll(List.of(game));
+
+    List<Hand> sortedHands = getHandsSortedByParticipation(game);
+    Hand handPlayer1 = sortedHands.get(0);
+    Hand handPlayer2 = sortedHands.get(1);
+    Hand handPlayer3 = sortedHands.get(2);
+
+    assertThat(handPlayer1.getNumberOfWonTricks(), is(1));
+    assertThat(handPlayer1.getPoints(), is(-1));
+    assertThat(handPlayer2.getNumberOfWonTricks(), is(0));
+    assertThat(handPlayer2.getPoints(), is(0));
+    assertThat(handPlayer3.getNumberOfWonTricks(), is(3));
+    assertThat(handPlayer3.getPoints(), is(-1));
+  }
+
+  // Note: No tests needed for last rounds stacked, because this scenario can not happen when number
+  // of won tricks are calculated, due to the implementation of adding rounds in case of last round
+  // getting stacked.
```

Comment:
Würde ich trotzdem testen, da 1. dann z.b gewisse spieler gar keine karte in der runde haben

Reply:
ah, du prüfst im CardEventHandler diese logik.
Dann würde ich schreiben: wird im CardEventHandlerTest getestet